superset-frontend/src/explore/components/controls/DateFilterControl/DateFilterControl.tsx

```diff
@@ -0,0 +1,872 @@
+  const { value = 'Last week', endpoints, onChange } = props;
+  const [actualTimeRange, setActualTimeRange] = useState<string>(value);
+
+  // State used for Modal
+  const [show, setShow] = useState<boolean>(false);
+  const [timeRangeFrame, setTimeRangeFrame] = useState<TimeRangeFrameType>(
+    guessTimeRangeFrame(value),
+  );
+  const [commonRange, setCommonRange] = useState<CommonRangeType>(
+    getDefaultOrCommonRange(value),
+  );
+  const [calendarRange, setCalendarRange] = useState<CalendarRangeType>(
+    getDefaultOrCalendarRange(value),
+  );
+  const [customRange, setCustomRange] = useState<CustomRangeType>(
+    customTimeRangeDecode(value).customRange,
+  );
+  const [advancedRange, setAdvancedRange] = useState<string>(
+    getAdvancedRange(value),
+  );
+  const [validTimeRange, setValidTimeRange] = useState<boolean>(false);
+  const [evalTimeRange, setEvalTimeRange] = useState<string>(value);
+
+  useEffect(() => {
+    fetchTimeRange(value, endpoints).then(({ value, error }) => {
+      if (error) {
+        setEvalTimeRange(error || '');
+        setValidTimeRange(false);
+      } else {
+        setActualTimeRange(value || '');
+        setValidTimeRange(true);
+      }
+    });
+  }, [value]);
+
+  useEffect(() => {
+    const value = getCurrentValue();
+    fetchTimeRange(value, endpoints).then(({ value, error }) => {
+      if (error) {
+        setEvalTimeRange(error || '');
+        setValidTimeRange(false);
+      } else {
+        setEvalTimeRange(value || '');
+        setValidTimeRange(true);
+      }
+    });
+  }, [timeRangeFrame, commonRange, calendarRange, customRange]);
+
+  function getCurrentValue(): string {
+    // get current time_range string
+    let value = 'Last week';
+    if (timeRangeFrame === 'Common') {
+      value = commonRange;
+    }
+    if (timeRangeFrame === 'Calendar') {
+      value = calendarRange;
+    }
+    if (timeRangeFrame === 'Custom') {
+      value = customTimeRangeEncode(customRange);
+    }
+    if (timeRangeFrame === 'Advanced') {
+      value = advancedRange;
+    }
+    if (timeRangeFrame === 'No Filter') {
+      value = 'No filter';
+    }
+    return value;
+  }
+
+  function getDefaultOrCommonRange(value: any): CommonRangeType {
+    const commonRange: CommonRangeType[] = [
+      'Last day',
+      'Last week',
+      'Last month',
+      'Last quarter',
+      'Last year',
+    ];
+    return commonRange.includes(value) ? value : 'Last week';
+  }
+
+  function getDefaultOrCalendarRange(value: any): CalendarRangeType {
+    const CalendarRange: CalendarRangeType[] = [
+      PreviousCalendarWeek,
+      PreviousCalendarMonth,
+      PreviousCalendarYear,
+    ];
+    return CalendarRange.includes(value) ? value : PreviousCalendarWeek;
+  }
+
+  function getAdvancedRange(value: string): string {
+    let since = '';
+    let until = '';
+    if (value.includes(SEPARATOR)) {
+      [since, until] = [...value.split(SEPARATOR)];
```

Comment:
When value has `SEPARATOR`, the current code split, then re-join to return the original string. 

How about

```ts
if (value.includes(SEPARATOR)) {
  return value;
} 
if (value.startsWith('Last')) {
  return [value, ''].join(SEPARATOR);
}
if (value.startsWith('Next')) {
  return ['', value].join(SEPARATOR);
}
return SEPARATOR;
```